event_routing_backends/handlers.py

```diff
@@ -0,0 +1,57 @@
+"""
```

Comment:
@Ian2012 I'm wondering why we have this piece of code(handlers.py) here. I think should be somewhere inside `edx-platform` or inside the event-bus plugin. Event routing backend should not handle/deserilize/forward events other than xAPI or Caliper backends.

Reply:
This code receives the signal from the event bus and then forwards the xAPI or Caliper events resulting from the tracking logs, to the configured backends.

I don't think this code should be in `edx-platform` as it would require `event-routing-backends` as a dependency and wouldn't allow compatibility with multiple Open edX versions. Also, this is the consumer code of a signal that mainly `event-routing-backends` is interested in, to transform tracking logs. I don't why it shouldn't be here.

Reply:
What happens is that the flow now will be:
`edx-platform -> event-tracking -> event-bus (redis) -> custom lms deployment run with ./manage.py consume events ...`

Events will be consumed in a different deployment and not in the same LMS thread as the request was made (or the tracking log was emitted). This would allow us to transform the tracking logs to xAPI or Caliper events with lower latency and with a dedicated service that can be fine-tuned for it.

The `openedx-events` signals are different than usual signals and the way we are emitting this signal in the `event-tracking` app is what allows us to emit the signal in the LMS process and receive it in an specific one.

Reply:
What if we follow this flow?
To produce events
`edx-platform -> event-tracking(EventBusBackend) -> event-bus (redis producer)`
To consume events
`event-bus (redis consumer) -> event-tracking(RoutingBackend) -> event-routing-backend`

Reply:
And even if that's implemented, that would just remove the signal references in this repository, but the code would be the same

Reply:
Oh sorry, I see what you mean. I was a bit confused with the concerns. You are right, the handler code is completely scoped to the event-tracking code and should be here.